### ISSUE TITLE
Can't delete my toots because of "t is undefined"

### ISSUE THREAD

**Issue by Sp3r4z** [tootsuite/mastodon#4928]
When I want to delete one of my toot, I click on "delete" link and nothing happens, the "confirmation toolbox" don't come, and Firefox (52.3.0) devtools give me an error message "TypeError: t is undefined".
Screen picture here: ![error message](https://framapic.org/qV1c6r3Hachp/GwYMHmXxHOoT.png)

PS: when I reload the page, it's "fix" the issue and I can delete my toot. But, some minutes later, problem still here… 

My instance is: Mastodon.xyz, it's v1.6.0

Hope that can help you fixing issue.

**Comment by TheKinrar** [comment]
Just checked I can delete my own toots so it doesn't look like an instance-related problem.

**Comment by Sp3r4z** [comment]
After some tests, with and without delete confirmation: problem doesn't appear in both situation…

It's not the first time, since v1.6.0 update, and always same problem about "t is undefined". I can't really explain why… Very curious and strange issue.

**Comment by syui** [comment]
I have the same problem. (Can't delete my toots because)

delete : https://mstdn.syui.cf/api/v1/statuses/473

```sh
# Delete Toot !
$ url=https://mstdn.syui.cf/api/v1/statuses/473
$ curl -X DELETE $url -H "Authorization: Bearer $access_token" -H "Content-Type: application/json"
> {}

# Check Toot ! (Can't delete my toots because)
$ curl -X GET $url -H "Authorization: Bearer $access_token" -H "Content-Type: application/json"
{"id":473,"created_at":"2017-09-15T22:30:08.603Z","in_reply_to_id":null,"in_reply_to_account_id":null,"sensitive":false,"spoiler_text":"","visibility":"public","language":"ja","uri":"http://mstdn.syui.cf/users/syui/statuses/473","content":"\u003cp\u003eup : master(472df245)\u003c/p\u003e","url":"http://mstdn.syui.cf/@syui/473","reblogs_count":0,"favourites_count":0,"favourited":false,"reblogged":false,"muted":false,"pinned":false,"reblog":null,"application":{"name":"test","website":null},"account":{"id":1,"username":"syui","acct":"syui","display_name":"","locked":false,"created_at":"2017-04-20T01:16:55.365Z","note":"\u003cp\u003e\u003ca href=\"https://syui.cf/\" rel=\"nofollow noopener\" target=\"_blank\"\u003e\u003cspan class=\"invisible\"\u003ehttps://\u003c/span\u003e\u003cspan class=\"\"\u003esyui.cf/\u003c/span\u003e\u003cspan class=\"invisible\"\u003e\u003c/span\u003e\u003c/a\u003e\u003c/p\u003e","url":"http://mstdn.syui.cf/@syui","avatar":"http://mstdn.syui.cf/system/accounts/avatars/000/000/001/original/db95035e82355278.png?1492749796","avatar_static":"http://mstdn.syui.cf/system/accounts/avatars/000/000/001/original/db95035e82355278.png?1492749796","header":"http://mstdn.syui.cf/system/accounts/headers/000/000/001/original/d95e96605d56ce5d.png?1492749796","header_static":"http://mstdn.syui.cf/system/accounts/headers/000/000/001/original/d95e96605d56ce5d.png?1492749796","followers_count":21,"following_count":1,"statuses_count":933},"media_attachments":[],"mentions":[],"tags":[]}
```

**Comment by syui** [comment]
```sh
$ curl -v -X DELETE ...

https://mstdn.syui.cf/api/v1/statuses/473
*   Trying 133.19.20.31...
* TCP_NODELAY set
* Connected to mstdn.syui.cf (133.19.20.31) port 443 (#0)
* TLS 1.2 connection using TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
* Server certificate: xxxxxxx.cloudflaressl.com
* Server certificate: COMODO ECC Domain Validation Secure Server CA 2
* Server certificate: COMODO ECC Certification Authority
> DELETE /api/v1/statuses/473 HTTP/1.1
> Host: mstdn.syui.cf
> User-Agent: curl/1.00
> Accept: */*
> Authorization: Bearer xxxxxxxxxxxxx
> Content-Type: application/json
>
< HTTP/1.1 200 OK
< Date: Sat, 16 Sep 2017 00:00:00 GMT
< Content-Type: application/json; charset=utf-8
< Transfer-Encoding: chunked
< Connection: keep-alive
< Set-Cookie: __cfduid=xxxxxxxx; expires=Sun, 16-Sep-18 00:00:00 GMT; path=/; domain=.syui.cf; HttpOnly
< X-Frame-Options: DENY
< X-Content-Type-Options: nosniff
< X-Xss-Protection: 1; mode=block
< X-Ratelimit-Limit: 300
< X-Ratelimit-Remaining: 299
< X-Ratelimit-Reset: 2017-09-16
< Vary: Accept-Encoding, Origin
< Etag: W/"xxxxxxx"
< Cache-Control: max-age=0, private, must-revalidate
< X-Request-Id: xxxxxxxx
< X-Runtime: 0.000
< Via: 1.1 vegur
< Strict-Transport-Security: max-age=0000000; includeSubDomains; preload
< Server: cloudflare-nginx
< CF-RAY: xxxx
<
* Connection #0 to host mstdn.syui.cf left intact
```

**Comment by syui** [comment]
I problem was solved.

What I did : 

- heroku buildpacks:remove https://github.com/rcaught/heroku-buildpack-cmake
- heroku repo:purge_cache -a $appname
- merge : master(efec5072)

**Comment by Sciss** [comment]
I have the same problem. It seems to have to do with 'cleaning up' modal dialogs. For example, I can reproduce this by clicking on content (image) of a toot, then closing the popup, then going to the account of the author of the toot, then selecting 'block' from the menu. Nothing happens.

To reproduce
- visit https://octodon.social/web/statuses/7096951
- click on image (sorry abount that)
- close poup
- go to the author's account: https://octodon.social/web/accounts/64695
- click on the menu button (horizontal lines)
- click 'Block @hichanomu'

Instead of a modal dialog opening, I get this JavaScript error:

```
TypeError: t is undefined[Learn More]  application-beb0139c24f52508abac.js:1:78001
	[651]/o</t.prototype.render https://octodon.social/packs/application-beb0139c24f52508abac.js:1:78001
	b._renderValidatedComponentWithoutOwnerOrContext https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:495688
	b._renderValidatedComponent https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:495812
	b.performInitialMount https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:491653
	b.mountComponent https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:490699
	i<.mountComponent https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:121178
	b.performInitialMount https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:491816
	b.mountComponent https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:490699
	i<.mountComponent https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:121178
	b.performInitialMount https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:491816
	b.mountComponent https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:490699
	i<.mountComponent https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:121178
	b.performInitialMount https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:491816
	b.mountComponent https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:490699
	i<.mountComponent https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:121178
	_<.Mixin.mountChildren https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:486564
	_.Mixin._createInitialChildren https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:475292
	_.Mixin.mountComponent https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:473506
	i<.mountComponent https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:121178
	_<.Mixin.mountChildren https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:486564
	_.Mixin._createInitialChildren https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:475292
	_.Mixin.mountComponent https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:473506
	i<.mountComponent https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:121178
	s.updateChildren https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:488532
	_<.Mixin._reconcilerUpdateChildren https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:486327
	_<.Mixin._updateChildren https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:487073
	_<.Mixin.updateChildren https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:486971
	_.Mixin._updateDOMChildren https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:477684
	_.Mixin.updateComponent https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:475890
	_.Mixin.receiveComponent https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:475443
	i<.receiveComponent https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:121572
	b._updateRenderedComponent https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:495163
	b._performComponentUpdate https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:494917
	b.updateComponent https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:494195
	b.performUpdateIfNecessary https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:493528
	i<.performUpdateIfNecessary https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:121754
	u https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:92987
	i.perform https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:136899
	i.perform https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:136899
	.perform https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:94040
	C https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:94217
	bound  self-hosted:915:17
	i.closeAll https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:137481
	i.perform https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:136979
	p.batchedUpdates https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:501972
	s https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:93200
	r https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:170404
	c<.enqueueSetState https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:171534
	r.prototype.setState https://octodon.social/packs/common-a62a87ca73b5d24af357.js:1:214353
	t/this.startAnimationIfNecessary/r.animationID< https://octodon.social/packs/application-beb0139c24f52508abac.js:1:69785
TypeError: t is undefined[Learn More]  application-beb0139c24f52508abac.js:1:78001
```

This is with octodon.social which is currently running 1.6.0 AFAIK. Browser is Firefox 52.3.0 (Linux Debian).

**Comment by Sciss** [comment]
Problem persists in 2.0.0rc3

**Comment by ProgVal** [comment]
I have a similar issue.

I am not very good with debugging Javascript, but using the "Pause on exception" feature of the Firefox debugger, I think the error comes from this line:

```
    const leftNav  = media.size > 1 && <button tabIndex='0' className='modal-container__nav modal-container__nav--left' onClick={this.handlePrevClick} aria-label={intl.formatMessage(messages.previous)}><i className='fa fa-fw fa-chevron-left' /></button>;
```

Here is the same line, in context:

```
  render () {
    const { media, intl, onClose } = this.props;

    const index = this.getIndex();

    const leftNav  = media.size > 1 && <button tabIndex='0' className='modal-container__nav modal-container__nav--left' onClick={this.handlePrevClick} aria-label={intl.formatMessage(messages.previous)}><i className='fa fa-fw fa-chevron-left' /></button>;
```

If my understanding of Javascript is correct, accessing `media.size` is equivalent to accessing `this.props.media.size`. However, `this.props.media` has no attribute `size`, only: `intl`, `onClose`, `onReblog`, `status`, an `proto`.

Here is a screenshot showing all of this:

![screenshot](https://user-images.githubusercontent.com/406946/32014193-a9b52a66-b9bd-11e7-96a6-7704c8e96334.png)

The same error+location happens if I click the boost button, but not the favorite button, so it is probably related to the popup.

Refreshing the page temporarily fixes the issue, but it happens again rather soon (several times today).


Mastodon version: 1.6.1 (oc.todon.fr), updated this week
Firefox version: 52.4

**Comment by ProgVal** [comment]
Fixed in v2.3.2.
